cmd/witness/cmd/root_test.go

```diff
@@ -0,0 +1,258 @@
+	}
+}
+
+func Test_loadSignersKeyPair(t *testing.T) {
+	privatePem, _ := rsakeypair(t)
+
+	keyOptions := options.KeyOptions{
+		KeyPath: privatePem.Name(),
+	}
+
+	_, errors := loadSigners(context.Background(), keyOptions)
+	if len(errors) != 0 {
+		t.Errorf("unexpected errors: %v", errors)
+	}
+
+	keyOptions.KeyPath = "not-a-file"
+	_, errors = loadSigners(context.Background(), keyOptions)
+	if len(errors) != 1 {
+		t.Errorf("expected 1 error, got %d", len(errors))
+	}
+}
+
+func Test_loadSignersCertificate(t *testing.T) {
+	_, intermediates, leafcert, leafkey := fullChain(t)
+
+	keyOptions := options.KeyOptions{
+		KeyPath: leafkey.Name(),
+		IntermediatePaths: []string{
+			intermediates[0].Name(),
+		},
+		CertPath: leafcert.Name(),
+	}
+
+	signers, errors := loadSigners(context.Background(), keyOptions)
+	if len(errors) != 0 {
+		t.Errorf("unexpected errors: %v", errors)
+	}
+
+	_, err := signers[0].Verifier()
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+	}
+
+	if len(signers) != 1 {
+		t.Errorf("expected 1 signer, got %d", len(signers))
+	}
+}
+
+func rsakeypair(t *testing.T) (privatePem *os.File, publicPem *os.File) {
+	privatekey, err := rsa.GenerateKey(rand.Reader, keybits)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	publickey := &privatekey.PublicKey
+
+	privatekey_bytes := x509.MarshalPKCS1PrivateKey(privatekey)
+
+	privateKeyBlock := &pem.Block{
+		Type:  "PRIVATE KEY",
+		Bytes: privatekey_bytes,
+	}
+
+	privatePem, err = os.CreateTemp("/tmp", "key.pem")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = pem.Encode(privatePem, privateKeyBlock)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	publicKeyBlock := &pem.Block{
+		Type:  "PUBLIC KEY",
+		Bytes: x509.MarshalPKCS1PublicKey(publickey),
+	}
+
+	publicPem, err = os.CreateTemp("/tmp", "key.pub")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = pem.Encode(publicPem, publicKeyBlock)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return privatePem, publicPem
+
+}
+
+//ref: https://jamielinux.com/docs/openssl-certificate-authority/appendix/intermediate-configuration-file.html
+func fullChain(t *testing.T) (caPem *os.File, intermediatePems []*os.File, leafPem *os.File, leafkeyPem *os.File) {
+	ca := &x509.Certificate{
+		SerialNumber: big.NewInt(42),
+		Subject: pkix.Name{
+			Organization: []string{"Witness Testing"},
+			CommonName:   "Witness Testing CA",
+		},
+		NotBefore:             time.Now(),
+		NotAfter:              time.Now().AddDate(10, 0, 0),
+		IsCA:                  true,
+		ExtKeyUsage:           []x509.ExtKeyUsage{x509.ExtKeyUsageCodeSigning},
+		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
+		BasicConstraintsValid: true,
+	}
+
+	caPrivateKey, err := rsa.GenerateKey(rand.Reader, keybits)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	caBytes, err := x509.CreateCertificate(rand.Reader, ca, ca, &caPrivateKey.PublicKey, caPrivateKey)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	caPem, err = os.CreateTemp("/tmp", "ca.pem")
```

Comment:
Use t.TempDir here : https://pkg.go.dev/testing#T.TempDir

Using paths such as `/tmp` is non-portable and our tests wouldn't run on Windows with this code.  Grab a TempDir from the testing.T struct and create your files there.  Each call to t.TempDir will return a new directory, so make sure to keep the path handy when you need files to exist in the same directory.

Go will automatically clean the test's temp directory up between runs.